akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2Multiplexer.scala

```diff
@@ -134,6 +133,14 @@ private[http2] trait Http2MultiplexerSupport { logic: GraphStageLogic with Stage
 
       private val controlFrameBuffer: mutable.Queue[FrameEvent] = new mutable.Queue[FrameEvent]
       private val sendableOutstreams: mutable.Queue[Int] = new mutable.Queue[Int]
+      def enqueueOutstream(streamId: Int): Unit = {
+        if (isDebugEnabled) require(!sendableOutstreams.contains(streamId), s"Stream [$streamId] was enqueued multiple times.") // requires expensive scanning -> avoid in production
+        sendableOutstreams.enqueue(streamId)
+      }
+      def removeOutStream(streamId: Int): Unit =
+        // in 2.12.x there's no Queue.-=, when 2.12.x support is dropped, this can be
+        // changed to Queue.-=
+        sendableOutstreams.dequeueAll(_ == streamId)
```

Comment:
is `.-=` much more efficient? in that case it might be worth it to add a `ccompat` helper to use `.-=` on 2.13 and delegate to `.dequeueAll(_ == streamId)` only on 2.12

Reply:
Oh wow, good that you asked. Turns out `dequeueAll` is not a side-effecting operation but it creates a copy of the queue, that is instantly discarded. So, there's probably another bug lurking there for which we currently have no tests...

Reply:
Ah, no, that's not what's happening. It's returning the elements it dequeued, of course...

Reply:
> is `.-=` much more efficient? in that case it might be worth it to add a `ccompat` helper to use `.-=` on 2.13 and delegate to `.dequeueAll(_ == streamId)` only on 2.12

:+1:, done

Reply:
I also tried to find a better solution for 2.12. It probably doesn't make lots of a difference and the idea that this is not common anyway.